pkg/agent/agent.go

```diff
@@ -378,11 +377,11 @@ func mergeResourceWithConfig() {
 
 	switch getAgentResourceType() {
 	case v1alpha1.DiscoveryAgentResourceName:
-		mergeDiscoveryAgentWithConfig(agent.cfg)
+		mergeDiscoveryAgentWithConfig(agent.cfg.(*config.CentralConfiguration))
```

Comment:
Can we check the type once before the switch statement?

Reply:
This is validated as central config prior to this usage.